<a href="https://colab.research.google.com/github/KumundzhievMaxim/AppliedDeepLearning/blob/main/EmotionRecognition/emotion_recognition.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Downloading the Data {When UNI Link unreacheable}

In [ ]:
# Audio Song 1-24
!wget -O Audio_Song_Actors_01-24.zip https://zenodo.org/record/1188976/files/Audio_Song_Actors_01-24.zip?download=1 

--2020-12-15 07:30:35--  https://zenodo.org/record/1188976/files/Audio_Song_Actors_01-24.zip?download=1
Resolving zenodo.org (zenodo.org)... 137.138.76.77
Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 225505317 (215M) [application/octet-stream]
Saving to: ‘Audio_Song_Actors_01-24.zip’

Audio_Song_Actors_0 100%[===================>] 215.06M  7.43MB/s    in 31s     

2020-12-15 07:31:07 (7.04 MB/s) - ‘Audio_Song_Actors_01-24.zip’ saved [225505317/225505317]



In [ ]:
# Audio Speech
!wget -O Audio_Speech_Actors_01-24.zip https://zenodo.org/record/1188976/files/Audio_Speech_Actors_01-24.zip?download=1

--2020-12-15 07:31:57--  https://zenodo.org/record/1188976/files/Audio_Speech_Actors_01-24.zip?download=1
Resolving zenodo.org (zenodo.org)... 137.138.76.77
Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 208468073 (199M) [application/octet-stream]
Saving to: ‘Audio_Speech_Actors_01-24.zip’

Audio_Speech_Actors 100%[===================>] 198.81M  8.33MB/s    in 27s     

2020-12-15 07:32:25 (7.41 MB/s) - ‘Audio_Speech_Actors_01-24.zip’ saved [208468073/208468073]



In [ ]:
# Unzip Audio Song 1-24 (./song)
!mkdir ./song && unzip /content/Audio_Song_Actors_01-24.zip -d ./song && rm -r /content/Audio_Song_Actors_01-24.zip

mkdir: cannot create directory ‘./song’: File exists


In [ ]:
# Unzip Audio Speech 1-24 (./speech)
!mkdir ./speech && unzip -q /content/Audio_Speech_Actors_01-24.zip -d ./speech  && rm -r /content/Audio_Speech_Actors_01-24.zip

In [ ]:
# Downalod Video Song Actor 1-9
import os

for video_index in range(1, 10):
  down_cmd = f'wget -O Video_Song_Actor_0{video_index}.zip https://zenodo.org/record/1188976/files/Video_Song_Actor_0{video_index}.zip?download=1'
  dir_cmd = f'mkdir ./video/Actor_0{video_index}'
  unzip_cmd = f'unzip -q Video_Song_Actor_0{video_index}.zip -d ./video/Actor_0{video_index}/'
  rm_cmd = f'rm -r Video_Song_Actor_0{video_index}.zip'

  print(f'Downloading Video_{video_index} zip')
  os.system(down_cmd)
  
  print('Created folder for particular Video zip')
  os.system(dir_cmd)
  
  print('Unzipping particular Video')
  os.system(unzip_cmd)

  print('Deleting particular Video zip')
  os.system(rm_cmd)


Created folder for particular Video zip
Unzipping particular Video
Deleting particular Video zip
Created folder for particular Video zip
Unzipping particular Video
Deleting particular Video zip
Created folder for particular Video zip
Unzipping particular Video
Deleting particular Video zip
Created folder for particular Video zip
Unzipping particular Video
Deleting particular Video zip


# ENTRY POINT

## Data Description
The RAVDESS contains **7356** files. 
Each file was rated **10 times** on emotional validity, intensity, and genuineness

[Video example](https://www.youtube.com/watch?v=Y7OQoNEu3dY)

The database contains 24 professional actors (12 female, 12 male), vocalizing two lexically-matched statements in a neutral North American accent. Speech includes calm, happy, sad, angry, fearful, surprise, and disgust expressions.


All conditions are available in three modality formats: Audio-only (16bit, 48kHz .wav), Audio-Video (720p H.264, AAC 48kHz, .mp4), and Video-only (no sound).  <b>Note, there are no song files for Actor_18.</b>


### Names convetions and details
#### Audio-only files {Audio_Speech_Actors}
Audio-only files of all actors (01-24) are available as two separate zip files (~200 MB each):

Speech file (Audio_Speech_Actors_01-24.zip, 215 MB) contains 1440 files: 60 trials per actor x 24 actors = 1440. 

#### Audio-Visual and Video-only files files {Video_Speech_Actor}
Speech files (Video_Speech_Actor_01.zip to Video_Speech_Actor_24.zip) collectively contains 2880 files: 60 trials per actor x 2 modalities (AV, VO) x 24 actors = 2880.


### Download Data when UNI Link reachable

In [ ]:
!wget -O ravdess.zip http://nipg1.inf.elte.hu:8765/ravdess.zip 

--2020-12-15 14:52:48--  http://nipg1.inf.elte.hu:8765/ravdess.zip
Resolving nipg1.inf.elte.hu (nipg1.inf.elte.hu)... 157.181.160.161
Connecting to nipg1.inf.elte.hu (nipg1.inf.elte.hu)|157.181.160.161|:8765... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13565818088 (13G) [application/zip]
Saving to: ‘ravdess.zip’

ravdess.zip         100%[===================>]  12.63G  16.2MB/s    in 11m 44s 

2020-12-15 15:04:33 (18.4 MB/s) - ‘ravdess.zip’ saved [13565818088/13565818088]



In [3]:
!unzip -q ravdess.zip 

## Download and Setup **jumcutter**
[Original Documentation](https://github.com/carykh/jumpcutter) 

In [4]:
!wget -O jumcutter.py https://raw.githubusercontent.com/carykh/jumpcutter/master/jumpcutter.py
!wget -O requirements.txt https://raw.githubusercontent.com/carykh/jumpcutter/master/requirements.txt
!pip install -r requirements.txt

--2020-12-15 17:08:04--  https://raw.githubusercontent.com/carykh/jumpcutter/master/jumpcutter.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8147 (8.0K) [text/plain]
Saving to: ‘jumcutter.py’

jumcutter.py        100%[===================>]   7.96K  --.-KB/s    in 0s      

2020-12-15 17:08:05 (63.2 MB/s) - ‘jumcutter.py’ saved [8147/8147]

--2020-12-15 17:08:05--  https://raw.githubusercontent.com/carykh/jumpcutter/master/requirements.txt
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 35 [text/plain]
Saving to: ‘requirements.txt’

req

In [23]:
import os
from tqdm import tqdm 
from typing import List
from pathlib import Path

def _unzip(path: List[str], type=None):
  if type == 'video':
    target = Path(ROOT_DIR, 'videos')
    target.mkdir(exist_ok=True)

  else:
    target = Path(ROOT_DIR, 'audio')
    target.mkdir(exist_ok=True)

  for video_index, video_zip in enumerate(path):  
    source = f'{ROOT_DIR}/{video_zip}'
    print(f'Unzip {source} TO {target}')

    unzip_cmd = f'unzip -q {source} -d {target} &' # umpersant relies for continue of the execution, otherwise STOPS. 
    os.system(unzip_cmd)

  return True

def delete_zips(paths: List[str]):
  """Permanently delete zips.
  """
  assert [False for path in paths if not path.endswith('*.zip')], 'paths have NOT .zip extensioned files.' 
  
  for path in paths:
    rm_cmd = f'rm -r {path} &' # umpersant relies for continue of the execution, otherwise STOPS. 
    os.system(rm_cmd)
    print(f'Permanently deleted {path}')
  return True


def filter_silenced_videos():
  """Remove the silence parts from the beginning
   and the end of video clips.

  Notes:
    Resulting Folder Structure for Particular Video 
    - Actor_01
      01-01-01-01-01-01.mp4
      ...
  Returns:
    Boolean True, whether all videos were preproccessed.  
  """

  video_folders_paths = os.listdir(ROOT_VIDEOS) # [Actor_01, ..., Actor_24]
  
  for _, video_folder in enumerate(tqdm(video_folders_paths)):
    _current = Path(ROOT_VIDEOS, video_folder) # [./ravdess/videos/Actor_<>/]
    _current_videos = os.listdir(_current) # [./ravdess/videos/Actor_<>/01-01...-.mp4, ...]
    
    for video_index, video in enumerate(tqdm(_current_videos)):
      print(f'Processing {video}, {video_index+1}/{len(_current_videos)}')
      # get rid of silience in the beggining and end of the video
      video_silence_cmd = f'python jumcutter.py --input_file {video} --output_file {video} --sounded_speed 1 --silent_speed 999999 --frame_margin 2 --frame_rate 30'
      os.system(video_silence_cmd)
  return True

In [24]:
# -- main

ROOT_DIR = './ravdess'
ROOT_VIDEOS = './ravdess/videos'

video_zips = sorted([zip_name for zip_name in os.listdir(ROOT_DIR) if zip_name.startswith('Video')])  
 
TOY_TARGET = video_zips[0]

_unzip([TOY_TARGET], type='video')
filter_silenced_videos()



  0%|          | 0/120 [00:00<?, ?it/s]

Unzip ./ravdess/Video_Speech_Actor_01.zip TO ravdess/videos
Processing 01-01-07-01-02-01-03.mp4, 1/120



  1%|          | 1/120 [00:02<05:01,  2.54s/it]

Processing 02-01-02-02-02-01-03.mp4, 2/120



  2%|▏         | 2/120 [00:02<03:43,  1.90s/it]

Processing 01-01-05-01-02-02-03.mp4, 3/120



  2%|▎         | 3/120 [00:03<02:49,  1.45s/it]

Processing 02-01-04-01-01-01-03.mp4, 4/120



  3%|▎         | 4/120 [00:03<02:12,  1.14s/it]

Processing 02-01-05-02-01-01-03.mp4, 5/120



  4%|▍         | 5/120 [00:04<01:46,  1.08it/s]

Processing 01-01-04-01-02-01-03.mp4, 6/120



  5%|▌         | 6/120 [00:04<01:28,  1.29it/s]

Processing 02-01-08-01-02-01-03.mp4, 7/120



  6%|▌         | 7/120 [00:05<01:15,  1.51it/s]

Processing 02-01-05-02-02-02-03.mp4, 8/120



  7%|▋         | 8/120 [00:05<01:06,  1.68it/s]

Processing 02-01-05-02-02-01-03.mp4, 9/120



  8%|▊         | 9/120 [00:05<01:00,  1.85it/s]

Processing 01-01-05-02-02-01-03.mp4, 10/120



  8%|▊         | 10/120 [00:06<00:55,  1.99it/s]

Processing 01-01-06-02-01-02-03.mp4, 11/120



  9%|▉         | 11/120 [00:06<00:51,  2.10it/s]

Processing 02-01-03-02-02-02-03.mp4, 12/120



 10%|█         | 12/120 [00:07<00:49,  2.18it/s]

Processing 01-01-07-02-01-02-03.mp4, 13/120



 11%|█         | 13/120 [00:07<00:48,  2.22it/s]

Processing 02-01-01-01-01-01-03.mp4, 14/120



 12%|█▏        | 14/120 [00:07<00:46,  2.26it/s]

Processing 01-01-02-01-01-01-03.mp4, 15/120



 12%|█▎        | 15/120 [00:08<00:45,  2.31it/s]

Processing 02-01-07-01-01-02-03.mp4, 16/120



 13%|█▎        | 16/120 [00:08<00:44,  2.32it/s]

Processing 01-01-02-01-01-02-03.mp4, 17/120



 14%|█▍        | 17/120 [00:09<00:44,  2.32it/s]

Processing 01-01-05-01-02-01-03.mp4, 18/120



 15%|█▌        | 18/120 [00:09<00:43,  2.33it/s]

Processing 02-01-07-02-02-01-03.mp4, 19/120



 16%|█▌        | 19/120 [00:10<00:43,  2.32it/s]

Processing 01-01-06-01-02-02-03.mp4, 20/120



 17%|█▋        | 20/120 [00:10<00:43,  2.30it/s]

Processing 02-01-05-01-02-02-03.mp4, 21/120



 18%|█▊        | 21/120 [00:10<00:42,  2.32it/s]

Processing 02-01-02-02-01-02-03.mp4, 22/120



 18%|█▊        | 22/120 [00:11<00:41,  2.35it/s]

Processing 01-01-02-02-02-01-03.mp4, 23/120



 19%|█▉        | 23/120 [00:11<00:41,  2.34it/s]

Processing 01-01-05-02-01-02-03.mp4, 24/120



 20%|██        | 24/120 [00:12<00:41,  2.33it/s]

Processing 02-01-06-02-02-01-03.mp4, 25/120



 21%|██        | 25/120 [00:12<00:41,  2.31it/s]

Processing 02-01-05-01-01-01-03.mp4, 26/120



 22%|██▏       | 26/120 [00:13<00:41,  2.27it/s]

Processing 02-01-02-02-02-02-03.mp4, 27/120



 22%|██▎       | 27/120 [00:13<00:40,  2.31it/s]

Processing 02-01-04-02-01-02-03.mp4, 28/120



 23%|██▎       | 28/120 [00:13<00:39,  2.31it/s]

Processing 02-01-01-01-01-02-03.mp4, 29/120



 24%|██▍       | 29/120 [00:14<00:39,  2.30it/s]

Processing 01-01-06-02-01-01-03.mp4, 30/120



 25%|██▌       | 30/120 [00:14<00:38,  2.31it/s]

Processing 01-01-02-01-02-01-03.mp4, 31/120



 26%|██▌       | 31/120 [00:15<00:38,  2.33it/s]

Processing 01-01-03-02-01-01-03.mp4, 32/120



 27%|██▋       | 32/120 [00:15<00:37,  2.35it/s]

Processing 01-01-03-02-02-02-03.mp4, 33/120



 28%|██▊       | 33/120 [00:16<00:36,  2.36it/s]

Processing 01-01-07-01-01-02-03.mp4, 34/120



 28%|██▊       | 34/120 [00:16<00:36,  2.38it/s]

Processing 02-01-08-01-02-02-03.mp4, 35/120



 29%|██▉       | 35/120 [00:16<00:35,  2.36it/s]

Processing 02-01-08-02-02-02-03.mp4, 36/120



 30%|███       | 36/120 [00:17<00:35,  2.38it/s]

Processing 01-01-02-02-02-02-03.mp4, 37/120



 31%|███       | 37/120 [00:17<00:34,  2.38it/s]

Processing 02-01-01-01-02-01-03.mp4, 38/120



 32%|███▏      | 38/120 [00:18<00:34,  2.38it/s]

Processing 01-01-03-01-01-02-03.mp4, 39/120



 32%|███▎      | 39/120 [00:18<00:34,  2.37it/s]

Processing 02-01-03-02-02-01-03.mp4, 40/120



 33%|███▎      | 40/120 [00:19<00:33,  2.36it/s]

Processing 01-01-08-02-01-01-03.mp4, 41/120



 34%|███▍      | 41/120 [00:19<00:33,  2.37it/s]

Processing 02-01-06-02-01-01-03.mp4, 42/120



 35%|███▌      | 42/120 [00:19<00:32,  2.37it/s]

Processing 01-01-08-01-01-01-03.mp4, 43/120



 36%|███▌      | 43/120 [00:20<00:32,  2.37it/s]

Processing 02-01-08-02-01-01-03.mp4, 44/120



 37%|███▋      | 44/120 [00:20<00:32,  2.34it/s]

Processing 02-01-05-01-02-01-03.mp4, 45/120



 38%|███▊      | 45/120 [00:21<00:32,  2.33it/s]

Processing 01-01-03-01-02-01-03.mp4, 46/120



 38%|███▊      | 46/120 [00:21<00:31,  2.36it/s]

Processing 01-01-01-01-02-02-03.mp4, 47/120



 39%|███▉      | 47/120 [00:22<00:31,  2.34it/s]

Processing 01-01-07-01-01-01-03.mp4, 48/120



 40%|████      | 48/120 [00:22<00:30,  2.35it/s]

Processing 02-01-03-01-01-02-03.mp4, 49/120



 41%|████      | 49/120 [00:22<00:29,  2.38it/s]

Processing 01-01-02-02-01-02-03.mp4, 50/120



 42%|████▏     | 50/120 [00:23<00:29,  2.37it/s]

Processing 02-01-07-02-01-02-03.mp4, 51/120



 42%|████▎     | 51/120 [00:23<00:28,  2.40it/s]

Processing 01-01-03-02-01-02-03.mp4, 52/120



 43%|████▎     | 52/120 [00:24<00:28,  2.39it/s]

Processing 01-01-04-02-01-01-03.mp4, 53/120



 44%|████▍     | 53/120 [00:24<00:28,  2.38it/s]

Processing 01-01-03-01-01-01-03.mp4, 54/120



 45%|████▌     | 54/120 [00:24<00:27,  2.38it/s]

Processing 02-01-06-02-02-02-03.mp4, 55/120



 46%|████▌     | 55/120 [00:25<00:27,  2.38it/s]

Processing 01-01-07-02-01-01-03.mp4, 56/120



 47%|████▋     | 56/120 [00:25<00:26,  2.41it/s]

Processing 01-01-04-01-02-02-03.mp4, 57/120



 48%|████▊     | 57/120 [00:26<00:26,  2.40it/s]

Processing 01-01-06-01-01-02-03.mp4, 58/120



 48%|████▊     | 58/120 [00:26<00:25,  2.41it/s]

Processing 01-01-08-01-02-02-03.mp4, 59/120



 49%|████▉     | 59/120 [00:27<00:25,  2.38it/s]

Processing 01-01-08-01-02-01-03.mp4, 60/120



 50%|█████     | 60/120 [00:27<00:25,  2.40it/s]

Processing 02-01-06-01-02-01-03.mp4, 61/120



 51%|█████     | 61/120 [00:27<00:24,  2.41it/s]

Processing 02-01-02-01-01-02-03.mp4, 62/120



 52%|█████▏    | 62/120 [00:28<00:24,  2.39it/s]

Processing 02-01-08-01-01-01-03.mp4, 63/120



 52%|█████▎    | 63/120 [00:28<00:23,  2.39it/s]

Processing 02-01-07-02-01-01-03.mp4, 64/120



 53%|█████▎    | 64/120 [00:29<00:23,  2.35it/s]

Processing 02-01-03-01-01-01-03.mp4, 65/120



 54%|█████▍    | 65/120 [00:29<00:23,  2.36it/s]

Processing 02-01-07-01-02-01-03.mp4, 66/120



 55%|█████▌    | 66/120 [00:29<00:22,  2.37it/s]

Processing 01-01-06-02-02-02-03.mp4, 67/120



 56%|█████▌    | 67/120 [00:30<00:22,  2.36it/s]

Processing 01-01-07-02-02-01-03.mp4, 68/120



 57%|█████▋    | 68/120 [00:30<00:21,  2.37it/s]

Processing 02-01-07-01-02-02-03.mp4, 69/120



 57%|█████▊    | 69/120 [00:31<00:21,  2.36it/s]

Processing 02-01-08-02-01-02-03.mp4, 70/120



 58%|█████▊    | 70/120 [00:31<00:21,  2.36it/s]

Processing 01-01-01-01-02-01-03.mp4, 71/120



 59%|█████▉    | 71/120 [00:32<00:21,  2.33it/s]

Processing 02-01-01-01-02-02-03.mp4, 72/120



 60%|██████    | 72/120 [00:32<00:20,  2.35it/s]

Processing 01-01-06-01-02-01-03.mp4, 73/120



 61%|██████    | 73/120 [00:32<00:19,  2.36it/s]

Processing 01-01-04-02-02-01-03.mp4, 74/120



 62%|██████▏   | 74/120 [00:33<00:19,  2.35it/s]

Processing 02-01-03-01-02-02-03.mp4, 75/120



 62%|██████▎   | 75/120 [00:33<00:18,  2.37it/s]

Processing 01-01-02-02-01-01-03.mp4, 76/120



 63%|██████▎   | 76/120 [00:34<00:18,  2.37it/s]

Processing 01-01-08-02-02-02-03.mp4, 77/120



 64%|██████▍   | 77/120 [00:34<00:18,  2.34it/s]

Processing 02-01-06-01-02-02-03.mp4, 78/120



 65%|██████▌   | 78/120 [00:35<00:17,  2.35it/s]

Processing 01-01-01-01-01-02-03.mp4, 79/120



 66%|██████▌   | 79/120 [00:35<00:17,  2.34it/s]

Processing 02-01-02-02-01-01-03.mp4, 80/120



 67%|██████▋   | 80/120 [00:35<00:17,  2.33it/s]

Processing 01-01-04-02-02-02-03.mp4, 81/120



 68%|██████▊   | 81/120 [00:36<00:16,  2.34it/s]

Processing 01-01-04-02-01-02-03.mp4, 82/120



 68%|██████▊   | 82/120 [00:36<00:16,  2.36it/s]

Processing 01-01-06-02-02-01-03.mp4, 83/120



 69%|██████▉   | 83/120 [00:37<00:15,  2.37it/s]

Processing 01-01-08-02-01-02-03.mp4, 84/120



 70%|███████   | 84/120 [00:37<00:15,  2.39it/s]

Processing 02-01-04-01-02-01-03.mp4, 85/120



 71%|███████   | 85/120 [00:38<00:14,  2.37it/s]

Processing 01-01-02-01-02-02-03.mp4, 86/120



 72%|███████▏  | 86/120 [00:38<00:14,  2.35it/s]

Processing 02-01-05-01-01-02-03.mp4, 87/120



 72%|███████▎  | 87/120 [00:38<00:13,  2.37it/s]

Processing 02-01-02-01-02-01-03.mp4, 88/120



 73%|███████▎  | 88/120 [00:39<00:13,  2.35it/s]

Processing 02-01-02-01-02-02-03.mp4, 89/120



 74%|███████▍  | 89/120 [00:39<00:13,  2.38it/s]

Processing 01-01-03-01-02-02-03.mp4, 90/120



 75%|███████▌  | 90/120 [00:40<00:12,  2.36it/s]

Processing 02-01-08-02-02-01-03.mp4, 91/120



 76%|███████▌  | 91/120 [00:40<00:12,  2.36it/s]

Processing 01-01-05-01-01-01-03.mp4, 92/120



 77%|███████▋  | 92/120 [00:41<00:11,  2.35it/s]

Processing 02-01-03-01-02-01-03.mp4, 93/120



 78%|███████▊  | 93/120 [00:41<00:11,  2.34it/s]

Processing 02-01-05-02-01-02-03.mp4, 94/120



 78%|███████▊  | 94/120 [00:41<00:10,  2.37it/s]

Processing 01-01-07-02-02-02-03.mp4, 95/120



 79%|███████▉  | 95/120 [00:42<00:10,  2.39it/s]

Processing 02-01-06-02-01-02-03.mp4, 96/120



 80%|████████  | 96/120 [00:42<00:10,  2.39it/s]

Processing 02-01-06-01-01-01-03.mp4, 97/120



 81%|████████  | 97/120 [00:43<00:09,  2.39it/s]

Processing 02-01-04-02-01-01-03.mp4, 98/120



 82%|████████▏ | 98/120 [00:43<00:09,  2.39it/s]

Processing 02-01-04-02-02-02-03.mp4, 99/120



 82%|████████▎ | 99/120 [00:43<00:08,  2.37it/s]

Processing 01-01-07-01-02-02-03.mp4, 100/120



 83%|████████▎ | 100/120 [00:44<00:08,  2.37it/s]

Processing 01-01-05-02-01-01-03.mp4, 101/120



 84%|████████▍ | 101/120 [00:44<00:08,  2.37it/s]

Processing 01-01-03-02-02-01-03.mp4, 102/120



 85%|████████▌ | 102/120 [00:45<00:07,  2.39it/s]

Processing 01-01-05-01-01-02-03.mp4, 103/120



 86%|████████▌ | 103/120 [00:45<00:07,  2.38it/s]

Processing 02-01-07-01-01-01-03.mp4, 104/120



 87%|████████▋ | 104/120 [00:46<00:06,  2.38it/s]

Processing 01-01-04-01-01-02-03.mp4, 105/120



 88%|████████▊ | 105/120 [00:46<00:06,  2.34it/s]

Processing 02-01-03-02-01-01-03.mp4, 106/120



 88%|████████▊ | 106/120 [00:46<00:05,  2.37it/s]

Processing 01-01-08-02-02-01-03.mp4, 107/120



 89%|████████▉ | 107/120 [00:47<00:05,  2.36it/s]

Processing 02-01-08-01-01-02-03.mp4, 108/120



 90%|█████████ | 108/120 [00:47<00:05,  2.37it/s]

Processing 01-01-04-01-01-01-03.mp4, 109/120



 91%|█████████ | 109/120 [00:48<00:04,  2.35it/s]

Processing 01-01-01-01-01-01-03.mp4, 110/120



 92%|█████████▏| 110/120 [00:48<00:04,  2.37it/s]

Processing 02-01-03-02-01-02-03.mp4, 111/120



 92%|█████████▎| 111/120 [00:49<00:03,  2.37it/s]

Processing 02-01-04-02-02-01-03.mp4, 112/120



 93%|█████████▎| 112/120 [00:49<00:03,  2.37it/s]

Processing 02-01-04-01-01-02-03.mp4, 113/120



 94%|█████████▍| 113/120 [00:49<00:02,  2.38it/s]

Processing 02-01-02-01-01-01-03.mp4, 114/120



 95%|█████████▌| 114/120 [00:50<00:02,  2.38it/s]

Processing 02-01-04-01-02-02-03.mp4, 115/120



 96%|█████████▌| 115/120 [00:50<00:02,  2.37it/s]

Processing 01-01-06-01-01-01-03.mp4, 116/120



 97%|█████████▋| 116/120 [00:51<00:01,  2.38it/s]

Processing 02-01-06-01-01-02-03.mp4, 117/120



 98%|█████████▊| 117/120 [00:51<00:01,  2.37it/s]

Processing 02-01-07-02-02-02-03.mp4, 118/120



 98%|█████████▊| 118/120 [00:51<00:00,  2.40it/s]

Processing 01-01-05-02-02-02-03.mp4, 119/120



 99%|█████████▉| 119/120 [00:52<00:00,  2.42it/s]

Processing 01-01-08-01-01-02-03.mp4, 120/120



  0%|          | 0/120 [00:00<?, ?it/s]

Processing 01-01-03-02-02-02-01.mp4, 1/120



  1%|          | 1/120 [00:00<00:50,  2.36it/s]

Processing 01-01-02-02-02-01-01.mp4, 2/120



  2%|▏         | 2/120 [00:00<00:50,  2.35it/s]

Processing 02-01-08-01-01-01-01.mp4, 3/120



  2%|▎         | 3/120 [00:01<00:49,  2.37it/s]

Processing 02-01-03-02-02-02-01.mp4, 4/120



  3%|▎         | 4/120 [00:01<00:49,  2.35it/s]

Processing 02-01-06-01-01-02-01.mp4, 5/120



  4%|▍         | 5/120 [00:02<00:48,  2.36it/s]

Processing 01-01-02-01-01-02-01.mp4, 6/120



  5%|▌         | 6/120 [00:02<00:47,  2.38it/s]

Processing 01-01-06-02-01-02-01.mp4, 7/120



  6%|▌         | 7/120 [00:02<00:47,  2.38it/s]

Processing 01-01-02-02-01-02-01.mp4, 8/120



  7%|▋         | 8/120 [00:03<00:47,  2.37it/s]

Processing 01-01-05-01-01-02-01.mp4, 9/120



  8%|▊         | 9/120 [00:03<00:47,  2.36it/s]

Processing 01-01-07-01-01-01-01.mp4, 10/120



  8%|▊         | 10/120 [00:04<00:46,  2.35it/s]

Processing 01-01-03-02-01-01-01.mp4, 11/120



  9%|▉         | 11/120 [00:04<00:46,  2.35it/s]

Processing 01-01-03-02-01-02-01.mp4, 12/120



 10%|█         | 12/120 [00:05<00:45,  2.35it/s]

Processing 02-01-08-02-02-01-01.mp4, 13/120



 11%|█         | 13/120 [00:05<00:45,  2.36it/s]

Processing 01-01-04-01-01-01-01.mp4, 14/120



 12%|█▏        | 14/120 [00:05<00:45,  2.34it/s]

Processing 02-01-08-01-01-02-01.mp4, 15/120



 12%|█▎        | 15/120 [00:06<00:45,  2.32it/s]

Processing 02-01-04-02-02-02-01.mp4, 16/120



 13%|█▎        | 16/120 [00:06<00:44,  2.34it/s]

Processing 02-01-03-01-01-02-01.mp4, 17/120



 14%|█▍        | 17/120 [00:07<00:43,  2.36it/s]

Processing 01-01-03-01-02-02-01.mp4, 18/120



 15%|█▌        | 18/120 [00:07<00:43,  2.36it/s]

Processing 02-01-07-02-01-01-01.mp4, 19/120



 16%|█▌        | 19/120 [00:08<00:43,  2.34it/s]

Processing 02-01-05-02-01-02-01.mp4, 20/120



 17%|█▋        | 20/120 [00:08<00:42,  2.35it/s]

Processing 01-01-08-01-01-01-01.mp4, 21/120



 18%|█▊        | 21/120 [00:08<00:42,  2.35it/s]

Processing 02-01-06-01-02-01-01.mp4, 22/120



 18%|█▊        | 22/120 [00:09<00:41,  2.36it/s]

Processing 01-01-04-01-01-02-01.mp4, 23/120



 19%|█▉        | 23/120 [00:09<00:40,  2.40it/s]

Processing 01-01-01-01-01-01-01.mp4, 24/120



 20%|██        | 24/120 [00:10<00:40,  2.37it/s]

Processing 02-01-05-01-01-01-01.mp4, 25/120



 21%|██        | 25/120 [00:10<00:39,  2.39it/s]

Processing 02-01-06-02-02-02-01.mp4, 26/120



 22%|██▏       | 26/120 [00:11<00:39,  2.37it/s]

Processing 02-01-08-02-01-02-01.mp4, 27/120



 22%|██▎       | 27/120 [00:11<00:39,  2.37it/s]

Processing 01-01-02-01-02-01-01.mp4, 28/120



 23%|██▎       | 28/120 [00:11<00:38,  2.36it/s]

Processing 01-01-05-02-02-01-01.mp4, 29/120



 24%|██▍       | 29/120 [00:12<00:38,  2.37it/s]

Processing 01-01-04-01-02-02-01.mp4, 30/120



 25%|██▌       | 30/120 [00:12<00:37,  2.39it/s]

Processing 01-01-06-01-02-02-01.mp4, 31/120



 26%|██▌       | 31/120 [00:13<00:37,  2.39it/s]

Processing 02-01-01-01-02-01-01.mp4, 32/120



 27%|██▋       | 32/120 [00:13<00:36,  2.40it/s]

Processing 02-01-07-02-02-01-01.mp4, 33/120



 28%|██▊       | 33/120 [00:13<00:36,  2.38it/s]

Processing 01-01-05-01-02-02-01.mp4, 34/120



 28%|██▊       | 34/120 [00:14<00:35,  2.39it/s]

Processing 02-01-07-01-02-01-01.mp4, 35/120



 29%|██▉       | 35/120 [00:14<00:35,  2.40it/s]

Processing 01-01-02-01-01-01-01.mp4, 36/120



 30%|███       | 36/120 [00:15<00:34,  2.40it/s]

Processing 01-01-05-02-02-02-01.mp4, 37/120



 31%|███       | 37/120 [00:15<00:34,  2.41it/s]

Processing 01-01-08-01-02-02-01.mp4, 38/120



 32%|███▏      | 38/120 [00:16<00:34,  2.39it/s]

Processing 01-01-03-01-01-02-01.mp4, 39/120



 32%|███▎      | 39/120 [00:16<00:33,  2.39it/s]

Processing 01-01-05-02-01-01-01.mp4, 40/120



 33%|███▎      | 40/120 [00:16<00:33,  2.38it/s]

Processing 02-01-04-01-01-02-01.mp4, 41/120



 34%|███▍      | 41/120 [00:17<00:33,  2.35it/s]

Processing 02-01-02-01-01-02-01.mp4, 42/120



 35%|███▌      | 42/120 [00:17<00:32,  2.37it/s]

Processing 02-01-02-02-01-01-01.mp4, 43/120



 36%|███▌      | 43/120 [00:18<00:32,  2.40it/s]

Processing 01-01-08-02-01-01-01.mp4, 44/120



 37%|███▋      | 44/120 [00:18<00:31,  2.42it/s]

Processing 02-01-03-02-01-02-01.mp4, 45/120



 38%|███▊      | 45/120 [00:18<00:30,  2.42it/s]

Processing 02-01-07-01-01-01-01.mp4, 46/120



 38%|███▊      | 46/120 [00:19<00:31,  2.38it/s]

Processing 01-01-06-01-01-02-01.mp4, 47/120



 39%|███▉      | 47/120 [00:19<00:30,  2.38it/s]

Processing 01-01-08-01-02-01-01.mp4, 48/120



 40%|████      | 48/120 [00:20<00:30,  2.36it/s]

Processing 01-01-08-01-01-02-01.mp4, 49/120



 41%|████      | 49/120 [00:20<00:29,  2.39it/s]

Processing 01-01-05-01-01-01-01.mp4, 50/120



 42%|████▏     | 50/120 [00:21<00:29,  2.40it/s]

Processing 01-01-04-02-01-01-01.mp4, 51/120



 42%|████▎     | 51/120 [00:21<00:28,  2.40it/s]

Processing 02-01-02-02-02-02-01.mp4, 52/120



 43%|████▎     | 52/120 [00:21<00:28,  2.40it/s]

Processing 02-01-03-01-02-01-01.mp4, 53/120



 44%|████▍     | 53/120 [00:22<00:28,  2.36it/s]

Processing 01-01-08-02-02-02-01.mp4, 54/120



 45%|████▌     | 54/120 [00:22<00:27,  2.36it/s]

Processing 01-01-04-02-02-01-01.mp4, 55/120



 46%|████▌     | 55/120 [00:23<00:27,  2.35it/s]

Processing 02-01-01-01-01-02-01.mp4, 56/120



 47%|████▋     | 56/120 [00:23<00:27,  2.36it/s]

Processing 01-01-02-02-02-02-01.mp4, 57/120



 48%|████▊     | 57/120 [00:24<00:26,  2.37it/s]

Processing 01-01-06-02-02-01-01.mp4, 58/120



 48%|████▊     | 58/120 [00:24<00:26,  2.37it/s]

Processing 02-01-02-01-02-01-01.mp4, 59/120



 49%|████▉     | 59/120 [00:24<00:25,  2.40it/s]

Processing 01-01-01-01-01-02-01.mp4, 60/120



 50%|█████     | 60/120 [00:25<00:24,  2.40it/s]

Processing 02-01-04-02-01-01-01.mp4, 61/120



 51%|█████     | 61/120 [00:25<00:24,  2.42it/s]

Processing 02-01-05-02-02-01-01.mp4, 62/120



 52%|█████▏    | 62/120 [00:26<00:23,  2.44it/s]

Processing 02-01-07-02-01-02-01.mp4, 63/120



 52%|█████▎    | 63/120 [00:26<00:23,  2.44it/s]

Processing 01-01-06-02-01-01-01.mp4, 64/120



 53%|█████▎    | 64/120 [00:26<00:22,  2.45it/s]

Processing 02-01-01-01-01-01-01.mp4, 65/120



 54%|█████▍    | 65/120 [00:27<00:22,  2.41it/s]

Processing 01-01-03-01-01-01-01.mp4, 66/120



 55%|█████▌    | 66/120 [00:27<00:22,  2.39it/s]

Processing 02-01-01-01-02-02-01.mp4, 67/120



 56%|█████▌    | 67/120 [00:28<00:21,  2.42it/s]

Processing 02-01-02-01-01-01-01.mp4, 68/120



 57%|█████▋    | 68/120 [00:28<00:21,  2.39it/s]

Processing 01-01-01-01-02-02-01.mp4, 69/120



 57%|█████▊    | 69/120 [00:28<00:21,  2.40it/s]

Processing 02-01-05-02-01-01-01.mp4, 70/120



 58%|█████▊    | 70/120 [00:29<00:21,  2.38it/s]

Processing 01-01-04-02-01-02-01.mp4, 71/120



 59%|█████▉    | 71/120 [00:29<00:20,  2.39it/s]

Processing 01-01-03-01-02-01-01.mp4, 72/120



 60%|██████    | 72/120 [00:30<00:19,  2.40it/s]

Processing 02-01-06-01-01-01-01.mp4, 73/120



 61%|██████    | 73/120 [00:30<00:19,  2.43it/s]

Processing 02-01-04-01-02-02-01.mp4, 74/120



 62%|██████▏   | 74/120 [00:31<00:19,  2.41it/s]

Processing 01-01-07-01-01-02-01.mp4, 75/120



 62%|██████▎   | 75/120 [00:31<00:18,  2.40it/s]

Processing 02-01-03-02-01-01-01.mp4, 76/120



 63%|██████▎   | 76/120 [00:31<00:18,  2.41it/s]

Processing 01-01-06-01-02-01-01.mp4, 77/120



 64%|██████▍   | 77/120 [00:32<00:17,  2.41it/s]

Processing 01-01-07-02-02-02-01.mp4, 78/120



 65%|██████▌   | 78/120 [00:32<00:17,  2.43it/s]

Processing 01-01-02-01-02-02-01.mp4, 79/120



 66%|██████▌   | 79/120 [00:33<00:16,  2.45it/s]

Processing 02-01-02-02-02-01-01.mp4, 80/120



 67%|██████▋   | 80/120 [00:33<00:16,  2.42it/s]

Processing 01-01-01-01-02-01-01.mp4, 81/120



 68%|██████▊   | 81/120 [00:33<00:16,  2.40it/s]

Processing 02-01-04-01-02-01-01.mp4, 82/120



 68%|██████▊   | 82/120 [00:34<00:15,  2.39it/s]

Processing 01-01-03-02-02-01-01.mp4, 83/120



 69%|██████▉   | 83/120 [00:34<00:15,  2.39it/s]

Processing 02-01-07-02-02-02-01.mp4, 84/120



 70%|███████   | 84/120 [00:35<00:15,  2.40it/s]

Processing 02-01-06-01-02-02-01.mp4, 85/120



 71%|███████   | 85/120 [00:35<00:14,  2.39it/s]

Processing 01-01-07-01-02-01-01.mp4, 86/120



 72%|███████▏  | 86/120 [00:36<00:14,  2.39it/s]

Processing 01-01-06-02-02-02-01.mp4, 87/120



 72%|███████▎  | 87/120 [00:36<00:13,  2.38it/s]

Processing 01-01-04-01-02-01-01.mp4, 88/120



 73%|███████▎  | 88/120 [00:36<00:13,  2.37it/s]

Processing 02-01-05-01-02-01-01.mp4, 89/120



 74%|███████▍  | 89/120 [00:37<00:13,  2.37it/s]

Processing 01-01-08-02-01-02-01.mp4, 90/120



 75%|███████▌  | 90/120 [00:37<00:12,  2.37it/s]

Processing 02-01-05-01-02-02-01.mp4, 91/120



 76%|███████▌  | 91/120 [00:38<00:12,  2.34it/s]

Processing 01-01-05-02-01-02-01.mp4, 92/120



 77%|███████▋  | 92/120 [00:38<00:11,  2.36it/s]

Processing 02-01-08-02-02-02-01.mp4, 93/120



 78%|███████▊  | 93/120 [00:39<00:11,  2.37it/s]

Processing 02-01-02-01-02-02-01.mp4, 94/120



 78%|███████▊  | 94/120 [00:39<00:11,  2.35it/s]

Processing 02-01-06-02-01-01-01.mp4, 95/120



 79%|███████▉  | 95/120 [00:39<00:10,  2.36it/s]

Processing 02-01-08-02-01-01-01.mp4, 96/120



 80%|████████  | 96/120 [00:40<00:10,  2.38it/s]

Processing 01-01-08-02-02-01-01.mp4, 97/120



 81%|████████  | 97/120 [00:40<00:09,  2.38it/s]

Processing 01-01-07-02-01-02-01.mp4, 98/120



 82%|████████▏ | 98/120 [00:41<00:09,  2.42it/s]

Processing 01-01-07-02-02-01-01.mp4, 99/120



 82%|████████▎ | 99/120 [00:41<00:08,  2.42it/s]

Processing 01-01-04-02-02-02-01.mp4, 100/120



 83%|████████▎ | 100/120 [00:41<00:08,  2.40it/s]

Processing 01-01-07-01-02-02-01.mp4, 101/120



 84%|████████▍ | 101/120 [00:42<00:07,  2.41it/s]

Processing 02-01-06-02-01-02-01.mp4, 102/120



 85%|████████▌ | 102/120 [00:42<00:07,  2.40it/s]

Processing 01-01-02-02-01-01-01.mp4, 103/120



 86%|████████▌ | 103/120 [00:43<00:07,  2.41it/s]

Processing 02-01-03-01-01-01-01.mp4, 104/120



 87%|████████▋ | 104/120 [00:43<00:06,  2.39it/s]

Processing 01-01-07-02-01-01-01.mp4, 105/120



 88%|████████▊ | 105/120 [00:44<00:06,  2.40it/s]

Processing 02-01-06-02-02-01-01.mp4, 106/120



 88%|████████▊ | 106/120 [00:44<00:05,  2.40it/s]

Processing 02-01-02-02-01-02-01.mp4, 107/120



 89%|████████▉ | 107/120 [00:44<00:05,  2.39it/s]

Processing 02-01-05-02-02-02-01.mp4, 108/120



 90%|█████████ | 108/120 [00:45<00:05,  2.38it/s]

Processing 02-01-08-01-02-01-01.mp4, 109/120



 91%|█████████ | 109/120 [00:45<00:04,  2.35it/s]

Processing 02-01-08-01-02-02-01.mp4, 110/120



 92%|█████████▏| 110/120 [00:46<00:04,  2.39it/s]

Processing 01-01-06-01-01-01-01.mp4, 111/120



 92%|█████████▎| 111/120 [00:46<00:03,  2.39it/s]

Processing 01-01-05-01-02-01-01.mp4, 112/120



 93%|█████████▎| 112/120 [00:46<00:03,  2.41it/s]

Processing 02-01-04-02-02-01-01.mp4, 113/120



 94%|█████████▍| 113/120 [00:47<00:02,  2.38it/s]

Processing 02-01-07-01-02-02-01.mp4, 114/120



 95%|█████████▌| 114/120 [00:47<00:02,  2.40it/s]

Processing 02-01-07-01-01-02-01.mp4, 115/120



 96%|█████████▌| 115/120 [00:48<00:02,  2.40it/s]

Processing 02-01-05-01-01-02-01.mp4, 116/120



 97%|█████████▋| 116/120 [00:48<00:01,  2.35it/s]

Processing 02-01-04-01-01-01-01.mp4, 117/120



 98%|█████████▊| 117/120 [00:49<00:01,  2.40it/s]

Processing 02-01-03-01-02-02-01.mp4, 118/120



 98%|█████████▊| 118/120 [00:49<00:00,  2.42it/s]

Processing 02-01-04-02-01-02-01.mp4, 119/120



 99%|█████████▉| 119/120 [00:49<00:00,  2.38it/s]

Processing 02-01-03-02-02-01-01.mp4, 120/120



  0%|          | 0/120 [00:00<?, ?it/s]

Processing 02-01-08-01-01-01-02.mp4, 1/120



  1%|          | 1/120 [00:00<00:50,  2.33it/s]

Processing 01-01-01-01-01-01-02.mp4, 2/120



  2%|▏         | 2/120 [00:00<00:50,  2.35it/s]

Processing 02-01-06-01-01-02-02.mp4, 3/120



  2%|▎         | 3/120 [00:01<00:50,  2.33it/s]

Processing 02-01-05-02-01-01-02.mp4, 4/120



  3%|▎         | 4/120 [00:01<00:49,  2.33it/s]

Processing 02-01-02-01-01-01-02.mp4, 5/120



  4%|▍         | 5/120 [00:02<00:49,  2.34it/s]

Processing 01-01-02-01-01-01-02.mp4, 6/120



  5%|▌         | 6/120 [00:02<00:49,  2.32it/s]

Processing 01-01-03-02-01-01-02.mp4, 7/120



  6%|▌         | 7/120 [00:03<00:48,  2.32it/s]

Processing 02-01-07-01-01-01-02.mp4, 8/120



  7%|▋         | 8/120 [00:03<00:48,  2.30it/s]

Processing 01-01-03-02-02-02-02.mp4, 9/120



  8%|▊         | 9/120 [00:03<00:47,  2.32it/s]

Processing 02-01-01-01-01-02-02.mp4, 10/120



  8%|▊         | 10/120 [00:04<00:47,  2.31it/s]

Processing 01-01-07-02-02-02-02.mp4, 11/120



  9%|▉         | 11/120 [00:04<00:46,  2.34it/s]

Processing 02-01-01-01-02-01-02.mp4, 12/120



 10%|█         | 12/120 [00:05<00:46,  2.33it/s]

Processing 02-01-06-01-01-01-02.mp4, 13/120



 11%|█         | 13/120 [00:05<00:46,  2.32it/s]

Processing 02-01-08-02-01-01-02.mp4, 14/120



 12%|█▏        | 14/120 [00:06<00:45,  2.33it/s]

Processing 01-01-02-01-02-02-02.mp4, 15/120



 12%|█▎        | 15/120 [00:06<00:45,  2.30it/s]

Processing 01-01-06-01-02-01-02.mp4, 16/120



 13%|█▎        | 16/120 [00:06<00:44,  2.32it/s]

Processing 01-01-05-01-01-01-02.mp4, 17/120



 14%|█▍        | 17/120 [00:07<00:43,  2.35it/s]

Processing 01-01-03-02-01-02-02.mp4, 18/120



 15%|█▌        | 18/120 [00:07<00:43,  2.36it/s]

Processing 01-01-03-01-01-02-02.mp4, 19/120



 16%|█▌        | 19/120 [00:08<00:43,  2.34it/s]

Processing 01-01-02-02-02-02-02.mp4, 20/120



 17%|█▋        | 20/120 [00:08<00:43,  2.31it/s]

Processing 02-01-04-02-01-02-02.mp4, 21/120



 18%|█▊        | 21/120 [00:09<00:42,  2.33it/s]

Processing 02-01-07-02-02-02-02.mp4, 22/120



 18%|█▊        | 22/120 [00:09<00:42,  2.33it/s]

Processing 02-01-04-01-01-01-02.mp4, 23/120



 19%|█▉        | 23/120 [00:09<00:42,  2.31it/s]

Processing 01-01-08-02-02-01-02.mp4, 24/120



 20%|██        | 24/120 [00:10<00:41,  2.30it/s]

Processing 01-01-08-01-01-01-02.mp4, 25/120



 21%|██        | 25/120 [00:10<00:41,  2.27it/s]

Processing 01-01-01-01-02-02-02.mp4, 26/120



 22%|██▏       | 26/120 [00:11<00:41,  2.28it/s]

Processing 02-01-08-01-01-02-02.mp4, 27/120



 22%|██▎       | 27/120 [00:11<00:40,  2.30it/s]

Processing 02-01-06-02-02-01-02.mp4, 28/120



 23%|██▎       | 28/120 [00:12<00:39,  2.31it/s]

Processing 01-01-04-02-01-01-02.mp4, 29/120



 24%|██▍       | 29/120 [00:12<00:39,  2.33it/s]

Processing 01-01-02-01-02-01-02.mp4, 30/120



 25%|██▌       | 30/120 [00:12<00:39,  2.30it/s]

Processing 02-01-02-02-01-01-02.mp4, 31/120



 26%|██▌       | 31/120 [00:13<00:38,  2.29it/s]

Processing 02-01-05-02-01-02-02.mp4, 32/120



 27%|██▋       | 32/120 [00:13<00:38,  2.31it/s]

Processing 02-01-08-02-02-01-02.mp4, 33/120



 28%|██▊       | 33/120 [00:14<00:37,  2.31it/s]

Processing 01-01-05-02-01-02-02.mp4, 34/120



 28%|██▊       | 34/120 [00:14<00:37,  2.32it/s]

Processing 02-01-04-01-01-02-02.mp4, 35/120



 29%|██▉       | 35/120 [00:15<00:36,  2.33it/s]

Processing 02-01-02-02-01-02-02.mp4, 36/120



 30%|███       | 36/120 [00:15<00:36,  2.33it/s]

Processing 01-01-03-01-02-01-02.mp4, 37/120



 31%|███       | 37/120 [00:15<00:36,  2.30it/s]

Processing 02-01-04-02-02-01-02.mp4, 38/120



 32%|███▏      | 38/120 [00:16<00:35,  2.30it/s]

Processing 01-01-05-01-01-02-02.mp4, 39/120



 32%|███▎      | 39/120 [00:16<00:35,  2.26it/s]

Processing 01-01-06-02-02-01-02.mp4, 40/120



 33%|███▎      | 40/120 [00:17<00:35,  2.26it/s]

Processing 02-01-07-02-01-01-02.mp4, 41/120



 34%|███▍      | 41/120 [00:17<00:34,  2.29it/s]

Processing 01-01-04-01-02-02-02.mp4, 42/120



 35%|███▌      | 42/120 [00:18<00:34,  2.25it/s]

Processing 01-01-04-02-02-01-02.mp4, 43/120



 36%|███▌      | 43/120 [00:18<00:33,  2.28it/s]

Processing 01-01-01-01-01-02-02.mp4, 44/120



 37%|███▋      | 44/120 [00:19<00:33,  2.27it/s]

Processing 01-01-08-01-01-02-02.mp4, 45/120



 38%|███▊      | 45/120 [00:19<00:33,  2.27it/s]

Processing 01-01-05-02-02-01-02.mp4, 46/120



 38%|███▊      | 46/120 [00:19<00:32,  2.26it/s]

Processing 02-01-05-01-01-02-02.mp4, 47/120



 39%|███▉      | 47/120 [00:20<00:32,  2.22it/s]

Processing 01-01-05-01-02-01-02.mp4, 48/120



 40%|████      | 48/120 [00:20<00:32,  2.24it/s]

Processing 02-01-04-02-01-01-02.mp4, 49/120



 41%|████      | 49/120 [00:21<00:31,  2.24it/s]

Processing 02-01-06-02-01-01-02.mp4, 50/120



 42%|████▏     | 50/120 [00:21<00:30,  2.28it/s]

Processing 01-01-07-01-01-01-02.mp4, 51/120



 42%|████▎     | 51/120 [00:22<00:30,  2.29it/s]

Processing 01-01-06-01-01-02-02.mp4, 52/120



 43%|████▎     | 52/120 [00:22<00:29,  2.29it/s]

Processing 01-01-04-01-02-01-02.mp4, 53/120



 44%|████▍     | 53/120 [00:23<00:28,  2.32it/s]

Processing 02-01-06-02-01-02-02.mp4, 54/120



 45%|████▌     | 54/120 [00:23<00:28,  2.33it/s]

Processing 02-01-07-01-01-02-02.mp4, 55/120



 46%|████▌     | 55/120 [00:23<00:27,  2.33it/s]

Processing 01-01-06-02-01-02-02.mp4, 56/120



 47%|████▋     | 56/120 [00:24<00:27,  2.34it/s]

Processing 01-01-04-02-02-02-02.mp4, 57/120



 48%|████▊     | 57/120 [00:24<00:27,  2.33it/s]

Processing 01-01-04-02-01-02-02.mp4, 58/120



 48%|████▊     | 58/120 [00:25<00:26,  2.32it/s]

Processing 02-01-03-02-02-01-02.mp4, 59/120



 49%|████▉     | 59/120 [00:25<00:26,  2.33it/s]

Processing 02-01-05-02-02-01-02.mp4, 60/120



 50%|█████     | 60/120 [00:26<00:25,  2.34it/s]

Processing 02-01-04-01-02-02-02.mp4, 61/120



 51%|█████     | 61/120 [00:26<00:25,  2.33it/s]

Processing 02-01-02-02-02-01-02.mp4, 62/120



 52%|█████▏    | 62/120 [00:26<00:25,  2.32it/s]

Processing 02-01-05-01-02-02-02.mp4, 63/120



 52%|█████▎    | 63/120 [00:27<00:24,  2.32it/s]

Processing 01-01-02-02-02-01-02.mp4, 64/120



 53%|█████▎    | 64/120 [00:27<00:24,  2.32it/s]

Processing 01-01-03-01-01-01-02.mp4, 65/120



 54%|█████▍    | 65/120 [00:28<00:23,  2.32it/s]

Processing 01-01-08-02-01-02-02.mp4, 66/120



 55%|█████▌    | 66/120 [00:28<00:23,  2.33it/s]

Processing 02-01-08-01-02-01-02.mp4, 67/120



 56%|█████▌    | 67/120 [00:29<00:22,  2.32it/s]

Processing 02-01-08-02-01-02-02.mp4, 68/120



 57%|█████▋    | 68/120 [00:29<00:22,  2.28it/s]

Processing 01-01-06-01-01-01-02.mp4, 69/120



 57%|█████▊    | 69/120 [00:29<00:22,  2.28it/s]

Processing 02-01-03-02-02-02-02.mp4, 70/120



 58%|█████▊    | 70/120 [00:30<00:21,  2.28it/s]

Processing 02-01-03-01-02-01-02.mp4, 71/120



 59%|█████▉    | 71/120 [00:30<00:21,  2.32it/s]

Processing 02-01-06-02-02-02-02.mp4, 72/120



 60%|██████    | 72/120 [00:31<00:20,  2.32it/s]

Processing 01-01-07-01-02-02-02.mp4, 73/120



 61%|██████    | 73/120 [00:31<00:20,  2.27it/s]

Processing 01-01-08-01-02-02-02.mp4, 74/120



 62%|██████▏   | 74/120 [00:32<00:20,  2.21it/s]

Processing 02-01-07-01-02-02-02.mp4, 75/120



 62%|██████▎   | 75/120 [00:32<00:20,  2.18it/s]

Processing 02-01-03-01-02-02-02.mp4, 76/120



 63%|██████▎   | 76/120 [00:33<00:20,  2.20it/s]

Processing 01-01-07-02-02-01-02.mp4, 77/120



 64%|██████▍   | 77/120 [00:33<00:19,  2.18it/s]

Processing 01-01-06-01-02-02-02.mp4, 78/120



 65%|██████▌   | 78/120 [00:33<00:19,  2.20it/s]

Processing 02-01-03-02-01-01-02.mp4, 79/120



 66%|██████▌   | 79/120 [00:34<00:18,  2.22it/s]

Processing 02-01-02-01-02-01-02.mp4, 80/120



 67%|██████▋   | 80/120 [00:34<00:17,  2.25it/s]

Processing 02-01-03-02-01-02-02.mp4, 81/120



 68%|██████▊   | 81/120 [00:35<00:17,  2.25it/s]

Processing 01-01-05-02-01-01-02.mp4, 82/120



 68%|██████▊   | 82/120 [00:35<00:16,  2.27it/s]

Processing 01-01-07-01-01-02-02.mp4, 83/120



 69%|██████▉   | 83/120 [00:36<00:16,  2.29it/s]

Processing 01-01-06-02-02-02-02.mp4, 84/120



 70%|███████   | 84/120 [00:36<00:15,  2.31it/s]

Processing 02-01-08-01-02-02-02.mp4, 85/120



 71%|███████   | 85/120 [00:37<00:15,  2.33it/s]

Processing 01-01-02-02-01-01-02.mp4, 86/120



 72%|███████▏  | 86/120 [00:37<00:14,  2.29it/s]

Processing 02-01-05-02-02-02-02.mp4, 87/120



 72%|███████▎  | 87/120 [00:37<00:14,  2.29it/s]

Processing 02-01-01-01-02-02-02.mp4, 88/120



 73%|███████▎  | 88/120 [00:38<00:14,  2.24it/s]

Processing 01-01-01-01-02-01-02.mp4, 89/120



 74%|███████▍  | 89/120 [00:38<00:13,  2.28it/s]

Processing 02-01-02-02-02-02-02.mp4, 90/120



 75%|███████▌  | 90/120 [00:39<00:13,  2.28it/s]

Processing 01-01-05-02-02-02-02.mp4, 91/120



 76%|███████▌  | 91/120 [00:39<00:12,  2.31it/s]

Processing 02-01-07-02-01-02-02.mp4, 92/120



 77%|███████▋  | 92/120 [00:40<00:12,  2.33it/s]

Processing 01-01-02-02-01-02-02.mp4, 93/120



 78%|███████▊  | 93/120 [00:40<00:11,  2.34it/s]

Processing 01-01-07-02-01-02-02.mp4, 94/120



 78%|███████▊  | 94/120 [00:40<00:11,  2.33it/s]

Processing 01-01-03-01-02-02-02.mp4, 95/120



 79%|███████▉  | 95/120 [00:41<00:10,  2.33it/s]

Processing 01-01-08-02-02-02-02.mp4, 96/120



 80%|████████  | 96/120 [00:41<00:10,  2.34it/s]

Processing 02-01-01-01-01-01-02.mp4, 97/120



 81%|████████  | 97/120 [00:42<00:09,  2.33it/s]

Processing 02-01-03-01-01-02-02.mp4, 98/120



 82%|████████▏ | 98/120 [00:42<00:09,  2.36it/s]

Processing 02-01-04-02-02-02-02.mp4, 99/120



 82%|████████▎ | 99/120 [00:43<00:08,  2.35it/s]

Processing 02-01-05-01-02-01-02.mp4, 100/120



 83%|████████▎ | 100/120 [00:43<00:08,  2.34it/s]

Processing 02-01-08-02-02-02-02.mp4, 101/120



 84%|████████▍ | 101/120 [00:43<00:08,  2.35it/s]

Processing 02-01-07-02-02-01-02.mp4, 102/120



 85%|████████▌ | 102/120 [00:44<00:07,  2.34it/s]

Processing 01-01-08-01-02-01-02.mp4, 103/120



 86%|████████▌ | 103/120 [00:44<00:07,  2.35it/s]

Processing 02-01-04-01-02-01-02.mp4, 104/120



 87%|████████▋ | 104/120 [00:45<00:06,  2.32it/s]

Processing 01-01-07-01-02-01-02.mp4, 105/120



 88%|████████▊ | 105/120 [00:45<00:06,  2.31it/s]

Processing 01-01-04-01-01-02-02.mp4, 106/120



 88%|████████▊ | 106/120 [00:46<00:06,  2.31it/s]

Processing 01-01-03-02-02-01-02.mp4, 107/120



 89%|████████▉ | 107/120 [00:46<00:05,  2.28it/s]

Processing 01-01-05-01-02-02-02.mp4, 108/120



 90%|█████████ | 108/120 [00:46<00:05,  2.32it/s]

Processing 01-01-08-02-01-01-02.mp4, 109/120



 91%|█████████ | 109/120 [00:47<00:04,  2.32it/s]

Processing 01-01-07-02-01-01-02.mp4, 110/120



 92%|█████████▏| 110/120 [00:47<00:04,  2.32it/s]

Processing 02-01-05-01-01-01-02.mp4, 111/120



 92%|█████████▎| 111/120 [00:48<00:03,  2.31it/s]

Processing 02-01-07-01-02-01-02.mp4, 112/120



 93%|█████████▎| 112/120 [00:48<00:03,  2.31it/s]

Processing 01-01-04-01-01-01-02.mp4, 113/120



 94%|█████████▍| 113/120 [00:49<00:02,  2.35it/s]

Processing 02-01-06-01-02-01-02.mp4, 114/120



 95%|█████████▌| 114/120 [00:49<00:02,  2.36it/s]

Processing 01-01-06-02-01-01-02.mp4, 115/120



 96%|█████████▌| 115/120 [00:49<00:02,  2.37it/s]

Processing 02-01-06-01-02-02-02.mp4, 116/120



 97%|█████████▋| 116/120 [00:50<00:01,  2.36it/s]

Processing 02-01-03-01-01-01-02.mp4, 117/120



 98%|█████████▊| 117/120 [00:50<00:01,  2.36it/s]

Processing 02-01-02-01-02-02-02.mp4, 118/120



 98%|█████████▊| 118/120 [00:51<00:00,  2.34it/s]

Processing 02-01-02-01-01-02-02.mp4, 119/120



 99%|█████████▉| 119/120 [00:51<00:00,  2.31it/s]

Processing 01-01-02-01-01-02-02.mp4, 120/120



100%|██████████| 3/3 [02:35<00:00, 51.74s/it]


True

In [56]:
class Audio:
  def __init__(self):
    self.root = './ravdess'
    self.audio_root = './ravdess/audio'
    self.audio_unzipepd = './ravdess/audio/Audio_Speech_Actors_01-24'

  def unzip(self):
    """unzip audio zip and move to audio_root path.
    """
    Path(self.audio_root).mkdir(exist_ok=True)
    audio_zips = sorted([zip_name for zip_name in os.listdir(self.root) if zip_name.startswith('Audio_Speech')])  
    
    for audio_index, audio_zip in enumerate(audio_zips):  
      source = f'{self.root}/{audio_zip}'
      target = f'{self.audio_root}/{audio_zip}'

      print(f'Unzip {source} TO {target}')

      unzip_cmd = f'unzip -q {source} -d {target} &' # umpersant relies for continue of the execution, otherwise STOPS. 
      os.system(unzip_cmd)      
    return True

  def extract_audio(self):
    audio_actors = os.listdir(self.audio_unzipepd)

    for audio_actor in tqdm(audio_actors, desc="Extracting audio..."):
      print(f'Processing Speech {audio_actor}')
      tracks = os.listdir(audio_actor)
      print(f'Actor has {len(tracks)} records.')
      pass

# audio = Audio()
# audio.unzip()

class Video:
  def __init__(self):
    self.root = './ravdess'
    self.video_root = './ravdess/videos'
    self.extracted_audio_root = './ravdess/extracted_audio'
    self.cmd = 'ffmpeg-git-20191209-amd64-static/ffmpeg'

  def extract_audio(self):
    actors = os.listdir(self.video_root)     
    for actor in actors:
      print(f'Processing Videos for {actor}')
      actor_videos = os.listdir(Path(self.video_root, actor))
      print(f'{actor} has {len(actor_videos)} videos.')

      for video in actor_videos:
        print(f'Processing video {video} for {actor}')
        input = f'{self.video_root}/{actor}/{video}'
        output = f'{self.extracted_audio_root}/{video.replace(".mp4", ".wav")}'
        cmd = f'ffmpeg -i {input} -vn -acodec libvorbis {output} &'
        os.system(cmd)

video_hadnler = Video()        
video_hadnler.extract_audio()


Processing Videos for .ipynb_checkpoints
.ipynb_checkpoints has 0 videos.
Processing Videos for Actor_03
Actor_03 has 120 videos.
IN ./ravdess/videos/Actor_03/01-01-07-01-02-01-03.mp4
OU ./ravdess/extracted_audio/01-01-07-01-02-01-03.ogg
IN ./ravdess/videos/Actor_03/02-01-02-02-02-01-03.mp4
OU ./ravdess/extracted_audio/02-01-02-02-02-01-03.ogg
IN ./ravdess/videos/Actor_03/01-01-05-01-02-02-03.mp4
OU ./ravdess/extracted_audio/01-01-05-01-02-02-03.ogg
IN ./ravdess/videos/Actor_03/02-01-04-01-01-01-03.mp4
OU ./ravdess/extracted_audio/02-01-04-01-01-01-03.ogg
IN ./ravdess/videos/Actor_03/02-01-05-02-01-01-03.mp4
OU ./ravdess/extracted_audio/02-01-05-02-01-01-03.ogg
IN ./ravdess/videos/Actor_03/01-01-04-01-02-01-03.mp4
OU ./ravdess/extracted_audio/01-01-04-01-02-01-03.ogg
IN ./ravdess/videos/Actor_03/02-01-08-01-02-01-03.mp4
OU ./ravdess/extracted_audio/02-01-08-01-02-01-03.ogg
IN ./ravdess/videos/Actor_03/02-01-05-02-02-02-03.mp4
OU ./ravdess/extracted_audio/02-01-05-02-02-02-03.ogg
IN ./r